xds/internal/balancer/outlierdetection/config.go

```diff
@@ -177,8 +178,5 @@ func (lbc *LBConfig) Equal(lbc2 *LBConfig) bool {
 	if !lbc.SuccessRateEjection.Equal(lbc2.SuccessRateEjection) {
 		return false
 	}
-	if !lbc.FailurePercentageEjection.Equal(lbc2.FailurePercentageEjection) {
-		return false
-	}
-	return cmp.Equal(lbc.ChildPolicy, lbc2.ChildPolicy)
+	return lbc.FailurePercentageEjection.Equal(lbc2.FailurePercentageEjection)
```

Comment:
You shouldn't just ignore the child policy field.

Reply:
I think the problem is, you shouldn't use `*outlierdetection.LBConfig` as the field in `DiscoveryMechanism`. It should be a config message without the child policy, because the child policy is specified by another field. Then you won't need an `Equal` method for this `LBConfig` type

This is I believe a missing piece in the outlier detection gRFC.
It needs to also specify what the new `ClusterResolver` balancer's config should look like.